Dockerfile

```diff
@@ -1,12 +1,24 @@
 ARG BUILDARCH=amd64
 ARG TARGETARCH=amd64
 
+### Front-end common
+FROM node:16 AS front-end-common
+
+WORKDIR /app/ui-build
+COPY ui-common/package.json .
+COPY ui-common/package-lock.json .
+RUN npm i
+COPY ui-common .
+RUN npm run build
+RUN npm pack
+
 ### Front-end
 FROM node:16 AS front-end
 
 WORKDIR /app/ui-build
 
 COPY ui/package.json ui/package-lock.json ./
+COPY --from=front-end-common ["/app/ui-build/up9-mizu-common-1.0.138.tgz", "."]
```

Comment:
just to avoid confusion maybe version should be " 0.0.0" and not "1.0.138"

Reply:
and change the version in ui-common/packge.json ,too.
this file name wont be created ever again